src/rpc.rs

```diff
@@ -28,6 +29,10 @@ impl JsonRpcService {
                 let rpc = RpcSolImpl;
                 io.extend_with(rpc.to_delegate());
 
+                panic::set_hook(Box::new(move |_| {
```

Comment:
Wait but why do we panic in the first place?  Rather than a blind `unwrap()` somewhere could we just check for error?

Reply:
Fair question. We can do that for the unwrap in the solana rpc module (which is the panic in testnet-sanity build 959). But there are actually 2 potential panics here; the 2nd one (which appears in build 871) is in a thread spawned within the jsonrpc crate, which we do not touch.
The panic::set_hook was my attempt to handle both at once.

If it seems worthwhile, I could dig into the jsonrpc crate and make a contribution there instead.

Reply:
Interesting.  Yeah we could split this into two:  (1) a fix for our unwrap, and (2) let’s try to get that panic removed/cleaned-up upstream  — if there’s no response we can look at alternatives but hopefully it won’t get to that